src/Compilers/Core/Portable/Collections/DictionaryExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// The collection of extension methods for the <see cref="Dictionary{TKey, TValue}"/> type
+    /// </summary>
+    internal static class DictionaryExtensions
+    {
+        /// <summary>
+        /// If the given key is not found in the dictionary, add it with the given value and return the value.
+        /// Otherwise return the existing value associated with that key.
+        /// </summary>
+        public static TValue GetOrAdd<TKey, TValue>(
+            this Dictionary<TKey, TValue> dictionary,
+            TKey key,
+            TValue value)
+            where TKey : notnull
+        {
+            if (dictionary.TryGetValue(key, out var existingValue))
+            {
+                return existingValue;
+            }
+            else
+            {
+                dictionary.Add(key, value);
+                return value;
+            }
```

Comment:
```suggestion
            if (!dictionary.TryGetValue(key, out var existingValue))
            {
                existingValue = value;
                dictionary.Add(key, existingValue);
            }
            return existingValue;
```

Reply:
It took me a few minutes to figure out what your code was doing. I find it counterintuitive to use a pattern-variable as a mutable local where it isn't definitely assigned.  Also, the name doesn't really make sense that way.  I'd rather leave it as it is.

---
In reply to: [387411596](https://github.com/dotnet/roslyn/pull/42012#discussion_r387411596) [](ancestors = 387411596)

Reply:
Sounds good, not wed to it

Reply:
Alternatively, you could use `out value` to remove the pattern variable